apps/backend/src/modules/contributors/contributor.repository.ts

```diff
@@ -103,8 +103,13 @@ class ContributorRepository implements Repository {
 			)
 			.leftJoin("git_emails", "contributors.id", "git_emails.contributor_id")
 			.leftJoin("activity_logs", "git_emails.id", "activity_logs.git_email_id")
-			.leftJoin("projects", "activity_logs.project_id", "projects.id")
-			.where("projects.id", projectId)
+			.leftJoin("projects", "activity_logs.project_id", "projects.id");
+
+		if (projectIds) {
+			query.whereIn("projects.id", projectIds);
```

Comment:
This thing contains a bug. 

The bug is that after we joined tables and this `where` query executed, we filter out all the projects, which are not in the `projectIds` list. So this function will return list of contributors but all of the them will have `projects` fields containing only projects that are in `projectIds`, so we lost information about other projects contributors may be involved into.

Check [my commit](https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/516/commits/c28407ca1a6f3a71a457d5a0df366488eb5e4a96) to see how I solved it, and try to adjust it for your case.

Here is small example of what I mean.
Here is the table from what I'd like to select all the rows, containing job 'Sales'
![image](https://github.com/user-attachments/assets/4033b18a-b8d1-4717-a2cc-4e838a897359)
So here is expected result
![image](https://github.com/user-attachments/assets/1b8f80ae-b972-4ad8-b2c4-7eb4d639d590)
And here is what we get using `where` function
![image](https://github.com/user-attachments/assets/7129b46a-5580-44c9-b03c-d67a62aab252)

And here is the code for this. I used this online PostgreSQL [compiler](https://onecompiler.com/postgresql)
```
-- create
CREATE TABLE job (
  id SERIAL PRIMARY KEY,
  name VARCHAR NOT NULL UNIQUE
);

CREATE TABLE person (
  id SERIAL PRIMARY KEY,
  name VARCHAR NOT NULL UNIQUE
);

CREATE TABLE person_to_job (
  id SERIAL PRIMARY KEY,
  personId INTEGER,
  jobId INTEGER,
  FOREIGN KEY (jobId) REFERENCES job(id),
  FOREIGN KEY (personId) REFERENCES person(id),
  UNIQUE (personId, jobId)
);

-- insert
INSERT INTO job (name) VALUES ('Sales');
INSERT INTO job (name) VALUES ('Accounting');
INSERT INTO job (name) VALUES ('Freelance');

INSERT INTO person (name) VALUES ('Clark');
INSERT INTO person (name) VALUES ('Dave');
INSERT INTO person (name) VALUES ('Ava');

INSERT INTO person_to_job (personId, jobId) VALUES (1, 1);
INSERT INTO person_to_job (personId, jobId) VALUES (1, 2);
INSERT INTO person_to_job (personId, jobId) VALUES (1, 3);
INSERT INTO person_to_job (personId, jobId) VALUES (2, 2);
INSERT INTO person_to_job (personId, jobId) VALUES (3, 1);

-- fetch 
SELECT
  p.id,
  p.name,
  ARRAY_AGG(j.name) job
FROM person p
LEFT JOIN person_to_job pj ON pj.personId=p.id
LEFT JOIN job j ON j.id=pj.jobId
-- WHERE j.name = 'Sales'
GROUP BY p.id
-- HAVING 'Sales' = ANY(ARRAY_AGG(j.name))
```

Reply:
The main idea is to compute the amount of commits contributors have on related to user projects. We don't only have to get contributors from these projects, but also sum amounts of commits only of these projects, so I believe it's neccessary to filter exactly the way it is now. Am I missing something?

Simply saying, according to your example, when user have permission ONLY for `Sales` project, they wouldn't see the amount of commits from `Freelance` or else.

Am I missing something?